tests/nocover/test_float_shrinking.py

```diff
@@ -0,0 +1,36 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import pytest
+
+import hypothesis.strategies as st
+from tests.common.debug import minimal
+
+
+def test_shrinks_to_simple_floats():
+    assert minimal(st.floats(), lambda x: x > 1) == 1.5
```

Comment:
Why 1.5 instead of 2.0?

Reply:
Ideally I'd like it to be 2.0, but it turns out to be a massive massive pain in the arse to get both integer floats ordering as their integer parts and floats biasing towards simplicity. The compromise solution we've ended up with is that floats order first by their integer part, then by the simplicity of their fractional part, so `1.5` is considered simpler than `2.0` because of its smaller integer part even though it has a fraction and `2.0` doesn't

Reply:
I think our proposed sorting key is: `(has_fractional_part, integer_part, simplicity_of_fractional_part, fractional_part, sign)`?

Adding the `has_fractional_part` to the key should make it prefer 2.0 over 1.5, and while I'm not sure `fractional_part` is currently in there it should be to distinguish between 0.25 and 0.75.

(mere implementation difficulty is not a good enough reason 😜)

Reply:
> I think our proposed sorting key is: (has_fractional_part, integer_part, simplicity_of_fractional_part, fractional_part, sign)

It's certainly a reasonable sorting key.

> while I'm not sure fractional_part is currently in there it should be to distinguish between 0.25 and 0.75.

The current `simplicity_of_fractional_part` is a total order, so adding `fractional_part` in there doesn't do anything to the order.

0.25 and 0.75 are readily distinguished under the current scheme because they have negative exponents, which shrink towards positive exponents. 0.25 is 1 * 2^{-2} while 0.75 is 1.5 * 2^{-1}. -1 > -2, so 0.75 is simpler.

1.25 and 1.75 are distinguished because the fractional part key there is (0...)10 for 1.25 and (0...)11 for 1.75, so the former sorts lexically before the latter., while 1.5 would be (0...)01, so is better yet.

> (mere implementation difficulty is not a good enough reason 😜)

It's not so much a question of implementation difficulty (I passed through much hairier schemes before settling on this one) as difficulty of shrinking.

A single bit encoding `has_fractional_part` and rounding in its absence might work reasonably well though. I'll give that a try and see what happens.

Reply:
The verdict is that it does work reasonably well, but in testing it I found that our shrinker struggles a fair bit with shrinking large floats to exact integer boundaries, and that all of the things I can figure out to do to fix it are way too expensive.

I've pushed a change with the extra fractional bit, and will ponder the minimization problem some more but recommend not blocking this on that.